compiler.go

```diff
@@ -316,6 +320,18 @@ func (e errFailedToResolve) Unwrap() error {
 	return e.err
 }
 
+func (e *executor) hasOverrideDescriptorProto() bool {
```

Comment:
We do this as a way to cheaply detect if descriptors are overridden. That way, if the file provided is not overridden, we can avoid the cost of wrapping the descriptor in a `linker.File` (which is not terribly expensive, but also not free since it must create an index of all symbols in the file).